src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/DWMColorization.cs

```diff
@@ -5,48 +5,34 @@
 using Microsoft.Win32;
 using MS.Internal;
 using System.Runtime.InteropServices;
+using MS.Internal.WindowsRuntime.Windows.UI.ViewManagement;
 
 namespace System.Windows;
 internal static class DwmColorization
 {
+    private static Color _currentApplicationAccentColor = Color.FromArgb(0xff, 0x00, 0x78, 0xd4);
+
+    private static UISettings _uiSettings = null;
+
     /// <summary>
-    /// The Accent Color that is currently applied to the application.
+    ///     Gets the current application accent color.
     /// </summary>
-    private static Color _currentApplicationAccentColor = Color.FromArgb(255, 0, 120, 212);
-
     internal static Color CurrentApplicationAccentColor
     {
         get { return _currentApplicationAccentColor; }
     }
 
-    [DllImport("UXTheme.dll")]
-    private static extern int GetUserColorPreference(in IMMERSIVE_COLOR_PREFERENCE colorPreference, bool alwaysFalse);
-
-    [DllImport("UXTheme.dll")]
-    private static extern uint GetColorFromPreference(in IMMERSIVE_COLOR_PREFERENCE colorPreference, IMMERSIVE_COLOR_TYPE colorType, bool isHighContrastEnabled, IMMERSIVE_HC_CACHE_MODE cacheMode);
-
-    [StructLayout(LayoutKind.Sequential)]
-    private struct IMMERSIVE_COLOR_PREFERENCE
-    {
-        public uint crStartColor;
-        public uint crAccentColor;
-    }
-
-    private enum IMMERSIVE_COLOR_TYPE
+    internal static UISettings _UISettings
     {
-        IMCLR_SystemAccentLight1 = 5,
-        IMCLR_SystemAccentLight2 = 6,
-        IMCLR_SystemAccentLight3 = 7,
-        IMCLR_SystemAccentDark1 = 3,
-        IMCLR_SystemAccentDark2 = 2,
-        IMCLR_SystemAccentDark3 = 1,
-        IMCLR_SystemAccent = 4
-    }
+        get
+        {
+            if (_uiSettings == null)
+            {
+                _uiSettings = new UISettings();
```

Comment:
right so you are never disposing it actually

Reply:
Yeah, it will be disposed with the application close. In what scenarios re-initialization of _uiSettings will/should take place?

Reply:
This was merely an observation to see whether the disposable pattern is of any use. You were also originally worried about the native object going stale, but I guess that is no longer a concern here. You can close this comment.